### PR TITLE
Abstract complex logic into a new Model

### PR DESCRIPTION
**Pull request by brianhelba**
This should allow all important operations to be called internally from Python, as well as via the HTTP API.

### CODE REVIEW

**Comment by brianhelba** [comment]
@manthey Can you take a look at this? I'm having trouble setting up the tests locally, so I'm not sure exactly why they're failing.

File: server/rest.py [hunk 1]
```diff
@@ -116,196 +55,58 @@ def createTiles(self, item, params):
...
+    @access.public
+    def getTilesInfo(self, itemId, params):
+        if itemId == 'test':
+            item = 'test'
+            imageArgs = self._parseTestParams(params)
+        else:
+            item = self.model('item').load(
+                id=itemId, level=AccessType.READ,
+                user=self.getCurrentUser(), exc=True)
+            imageArgs = dict()
```
Comment (manthey) [inline server/rest.py]: This needs to be the params dict.  We support additional parameters for both getTilesInfo and getTile, such as asking for encoding='PNG' (the SVS tiles specifically support encoding, jpegSubsampling and jpegQuality as additional options).

File: server/rest.py [hunk 2]
```diff
@@ -330,11 +131,30 @@ def getTile(self, itemId, z, x, y, params):
...
-        cherrypy.response.headers['Content-Type'] = tileSource.getTileMimeType()
+        if itemId == 'test':
+            item = 'test'
+            imageArgs = self._parseTestParams(params)
+        else:
+            # TODO: cache the user / item loading too
+            item = self.model('item').load(
+                id=itemId, level=AccessType.READ,
+                user=self.getCurrentUser(), exc=True)
+            imageArgs = dict()
```
Comment (manthey) [inline server/rest.py]: This needs to be the params dict.  See above.

File: server/models/image_item.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            job = self._createLargeImageJob(item, fileObj, user, token)
+            item['expectedLargeImage'] = True
+            item['largeImageOriginalId'] = fileObj['_id']
+            item['largeImageJobId'] = job['_id']
+
+        self.save(item)
+        return job
+
+    def _createLargeImageJob(self, item, fileObj, user, token):
+        path = os.path.join(os.path.dirname(__file__), 'create_tiff.py')
```
Comment (manthey) [inline server/models/image_item.py]: This path is no longer correct.  create_tiff.py is in the parent directory, so you either need to move it or change this to `path = os.path.join(os.path.dirname(__file__), '..', 'create_tiff.py')`

**Comment by brianhelba** [comment]
@manthey PTAL.

**Comment by manthey** [comment]
@brianhelba I changed what exceptions are thrown, since we don't want Rest exceptions in the general model.  I'll merge this in; if you want changes, we'll do it as another PR.

**Comment by brianhelba** [comment]
This is perfect, thanks @manthey.

**Pull request merged**